RecoParticleFlow/PFProducer/plugins/importers/GeneralTracksImporterWithVeto.cc

```diff
@@ -0,0 +1,211 @@
+#include "RecoParticleFlow/PFProducer/interface/BlockElementImporterBase.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrackFwd.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrack.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "RecoParticleFlow/PFProducer/interface/PFMuonAlgo.h"
+#include "RecoParticleFlow/PFTracking/interface/PFTrackAlgoTools.h"
+
+class GeneralTracksImporterWithVeto : public BlockElementImporterBase {
+public:
+  GeneralTracksImporterWithVeto(const edm::ParameterSet& conf,
+		    edm::ConsumesCollector& sumes) :
+    BlockElementImporterBase(conf,sumes),
+    _src(sumes.consumes<reco::PFRecTrackCollection>(conf.getParameter<edm::InputTag>("source"))),
+    _veto(sumes.consumes<reco::PFRecTrackCollection>(conf.getParameter<edm::InputTag>("veto"))),
+    _muons(sumes.consumes<reco::MuonCollection>(conf.getParameter<edm::InputTag>("muonSrc"))),
+    _DPtovPtCut(conf.getParameter<std::vector<double> >("DPtOverPtCuts_byTrackAlgo")),
+    _NHitCut(conf.getParameter<std::vector<unsigned> >("NHitCuts_byTrackAlgo")),
+    _useIterTracking(conf.getParameter<bool>("useIterativeTracking")),
+    _cleanBadConvBrems(conf.existsAs<bool>("cleanBadConvertedBrems") ? conf.getParameter<bool>("cleanBadConvertedBrems") : false),
+    _debug(conf.getUntrackedParameter<bool>("debug",false)) {
```

Comment:
Variable names starting with "_" are explicitly forbidden by the CMS coding rules. They can be cleaned up in a subsequent PR. There's even a Wisconsin student who can take care of it if you don't have time.
https://cms-cpt-software.web.cern.ch/cms-cpt-software/General/CodingAndStyleRules.pdf